### ISSUE TITLE
Should valid-jsdoc check @private on exported functions?

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#8078]
Inspired by [this comment](https://github.com/eslint/eslint/pull/8069#issuecomment-279599689). (I know that has nothing to do with ES6 module exports, though.)

**What rule do you want to change?**
valid-jsdoc

**Does this change cause the rule to produce more or fewer warnings?**
More (behind an option)

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option: `{ "disallowPrivateExports": true }` (open to bikeshedding)

**Please provide some example code that this change will affect:**

```js
/**
 * Does the foo.
 * @private
 * @returns void
 */
export function foo() {
}
```

**What does the rule currently do for this code?**
No warning.

**What will the rule do after it's changed?**
Warn that an exported function should not have `@private` in its JSDoc.

**Comment by platinumazure** [comment]
Open questions:

* Is it worth adding support for Node modules as well here?
* Is it worth trying to use escope or similar to catch functions exported within an object (i.e., detect cases where the variable is exported separately from the function definition)?

**Comment by gyandeeps** [comment]
I am not against the idea but I think now we might be at a point where we are overloading the rule with lot of stuff, since the name is so generic `valid-jsdoc` (anything you don't like is invalid :smile: ). I wish there was a plugin for `jsdoc` where all `jsdoc` niche rules can live. similar to the work done for imports.

**Comment by ilyavolodin** [comment]
@gyandeeps [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc)?

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
